From: Tommi <tommi@webrtc.org>
Date: Fri, 23 Feb 2024 14:49:05 +0100
Subject: (cherry-pick-branch-heads/6312) [M123 MERGE] Demote RTC_CHECK for
 sctp_mid() to RTC_LOG(LS_ERROR) if unavailable

(cherry picked from commit efbfc40029b6986137f9179476955c263da7052a)

Bug: chromium:326275823, chromium:325900490
Change-Id: Icfb8850867d1e39f23661422693da4f2829ecc57
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/340460
Reviewed-by: Evan Shrubsole <eshr@webrtc.org>
Commit-Queue: Tomas Gunnarsson <tommi@webrtc.org>
Cr-Original-Commit-Position: refs/heads/main@{#41793}
No-Try: True
Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/342560
Reviewed-by: Guido Urdaneta <guidou@webrtc.org>
Commit-Queue: Florent Castelli <orphis@webrtc.org>
Cr-Commit-Position: refs/branch-heads/6312@{#3}
Cr-Branched-From: 0355f455a48b141a8277442825ec776a74d66fb7-refs/heads/main@{#41763}
---
 pc/sdp_offer_answer.cc | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/pc/sdp_offer_answer.cc b/pc/sdp_offer_answer.cc
index d75606ab4e..95d54978c7 100644
--- a/pc/sdp_offer_answer.cc
+++ b/pc/sdp_offer_answer.cc
@@ -4297,11 +4297,13 @@ void SdpOfferAnswerHandler::GetOptionsForUnifiedPlanOffer(
             GetMediaDescriptionOptionsForRejectedData(mid));
       } else {
         const auto data_mid = pc_->sctp_mid();
-        RTC_CHECK(data_mid);
-        if (mid == data_mid.value()) {
+        if (data_mid.has_value() && mid == data_mid.value()) {
           session_options->media_description_options.push_back(
               GetMediaDescriptionOptionsForActiveData(mid));
         } else {
+          if (!data_mid.has_value()) {
+            RTC_LOG(LS_ERROR) << "Datachannel transport not available: " << mid;
+          }
           session_options->media_description_options.push_back(
               GetMediaDescriptionOptionsForRejectedData(mid));
         }
